pypga/modules/migen/spimachine.py

```diff
@@ -127,7 +127,7 @@ def __init__(self, data_width=32, div_width=8):
         ]
         self.sync += [
                 If(reg.load,
-                    n.eq(self.length),
+                    n.eq(self.length - 1),
```

Comment:
Fixes wrong number of clock cycles. If you start from n and count down to zero, then you have n+1 clock cycles.